deepchem/utils/evaluate.py

```diff
@@ -2,122 +2,319 @@
 Utility functions to evaluate models on datasets.
 """
 import csv
+import logging
 import numpy as np
 import warnings
 import pandas as pd
 import sklearn
 from deepchem.utils.save import log
 from deepchem.trans import undo_transforms
 from deepchem.metrics import from_one_hot
+from deepchem.metrics import Metric
 
-__author__ = "Bharath Ramsundar"
-__copyright__ = "Copyright 2016, Stanford University"
-__license__ = "MIT"
+logger = logging.getLogger(__name__)
+
+
+def output_statistics(scores, stats_out):
+  """Write computed stats to file.
+
+  Statistics are written to specified `stats_out` file.
+
+  Parameters
+  ----------
+  scores: dict
+    Dictionary mapping names of metrics to scores.
+  stats_out: str
+    Name of file to write scores to.
+  """
+  logger.warning("output_statistics is deprecated.")
+  with open(stats_out, "w") as statsfile:
+    statsfile.write(str(scores) + "\n")
+
+
+def output_predictions(dataset, y_preds, csv_out):
+  """Writes predictions to file.
+
+  Writes predictions made on `dataset` to a specified file on
+  disk. `dataset.ids` are used to format predictions. The produce CSV file will have format as follows
+
+  | ID          | Task1Name    | Task2Name    |
+  | ----------- | ------------ | ------------ |
+  | identifer1  | prediction11 | prediction12 |
+  | identifer2  | prediction21 | prediction22 |
+
+  Parameters
+  ----------
+  dataset: dc.data.Dataset
+    Dataset on which predictions have been made.
+  y_preds: np.ndarray
+    Predictions to output
+  csv_out: str
+    Name of file to write predictions to.
+  """
+  mol_ids = dataset.ids
```

Comment:
Currently, we start supporting inorganic crystals. So, I think this variable name is not good.

Reply:
Good suggestion! Have swapped over to a new name